src/js/visual/image.js

```diff
@@ -358,6 +358,8 @@ ripe.Image.prototype.update = async function(state, options = {}) {
               )
             : {};
 
+        if (this._cancellingUpdates) return;

```

Comment:
Perhaps just add a comment explaining the reason for this line, since these functions are heavily documented and the logic of update is complex